core/src/main/java/io/crate/types/ObjectType.java

```diff
@@ -56,9 +59,21 @@ public String getName() {
     @Override
     @SuppressWarnings("unchecked")
     public Map<String, Object> value(Object value) {
+        if (value instanceof BytesRef) {
```

Comment:
I think the only case where we have strings is inside an object if it is returned as a whole.

E.g.  `select o from t` 

But I realized that I didn't register the cast function correctly - pushed again to the branch (squashed it already :( )